pyquil/experiment/_main.py

```diff
@@ -431,6 +443,48 @@ def build_symmetrization_memory_maps(
             memory_maps.append({f"{label}": list(zeros)})
         return memory_maps
 
+    def generate_calibration_experiment(self) -> "TomographyExperiment":
+        """
+        Generate another ``TomographyExperiment`` object that can be used to calibrate the various
+        multi-qubit observables involved in this ``TomographyExperiment``. This is achieved by
+        preparing the plus-one (minus-one) eigenstate of each ``out_operator``, and measuring the
+        resulting expectation value of the same ``out_operator``. Ideally, this would always give
+        +1 (-1), but when symmetric readout error is present the effect is to scale the resultant
+        expectations by some constant factor. Determining this scale factor is what we call
+        *readout calibration*, and then the readout error in subsequent measurements can then be
+        mitigated by simply dividing by the scale factor.
+
+        :return: A new ``TomographyExperiment`` that can calibrate the readout error of all the
+            observables involved in this experiment.
+        """
+        if self.calibration != CalibrationMethod.PLUS_EIGENSTATE:
+            raise ValueError('We currently only support the "plus eigenstate" calibration method.')
+
+        calibration_settings = []
+        for settings in self:
+            calibration_settings.append(
+                ExperimentSetting(
+                    in_state=settings[0].out_operator,
```

Comment:
Prefer to assert that settings is a length 1 list somewhere (e.g. if not here, then in the constructor).